assets/js/modules/analytics/components/dashboard/LegacyAdSenseDashboardWidgetTopPagesTableSmall.js

```diff
@@ -20,95 +20,126 @@
-		}
+			<Layout
+				className="googlesitekit-top-earnings-pages"
+				footer
+				footerCTALabel={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+				footerCTALink="http://analytics.google.com"
+				fill
+			>
+				{ component }
+			</Layout>
+		</div>
+	);
+}
 
-		const headers = [
-			{
-				title: __( 'Top Earning Pages', 'google-site-kit' ),
-				tooltip: __( 'Top Earning Pages', 'google-site-kit' ),
-				primary: true,
-			},
-			{
-				title: __( 'Revenue', 'google-site-kit' ),
-				tooltip: __( 'Revenue', 'google-site-kit' ),
-			},
-		];
+const LegacyAdSenseDashboardWidgetTopPagesTableSmall = ( { data } ) => {
+	const { startDate, endDate } = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
 
-		const links = [];
-		const dataMapped = data[ 0 ].data.rows.map( ( row, i ) => {
-			links[ i ] = row.dimensions[ 1 ];
-			return [
-				row.dimensions[ 0 ],
-				Number( row.metrics[ 0 ].values[ 0 ] ).toFixed( 2 ),
-			];
-		} );
+	const adsenseData = useSelect( ( select ) => select( MODULES_ADSENSE ).getReport( {
+		startDate,
+		endDate,
+		metrics: 'EARNINGS',
+	} ) );
 
-		const options = {
-			hideHeader: false,
-			chartsEnabled: false,
-			cap: 5,
-			links,
-		};
+	setCurrencyFormat( adsenseData );
+	if ( ! data || ! data.length ) {
+		return null;
+	}
+	const { rows } = data?.[ 0 ]?.data || {};
+	if ( ! Array.isArray( rows ) ) {
+		return null;
+	}
 
-		const dataTable = getDataTableFromData( dataMapped, headers, options );
+	// Before ReportTable, this originally used
+	// the DataTable's `cap` prop
+	const firstFiveRows = rows.slice( 0, 5 );
+
+	return renderLayout(
+		<TableOverflowContainer>
+			<ReportTable
+				rows={ firstFiveRows }
+				columns={ tableColumns }
+			/>
+		</TableOverflowContainer>
+	);
+};
 
-		return (
-			LegacyAdSenseDashboardWidgetTopPagesTableSmall.renderLayout(
-				<TableOverflowContainer>
-					{ dataTable }
-				</TableOverflowContainer>
-			)
-		);
-	}
-}
+const tableColumns = [
+	{
+		title: __( 'Top Earning Pages', 'google-site-kit' ),
+		tooltip: __( 'Top Earning Pages', 'google-site-kit' ),
+		primary: true,
+		Component: ( { row } ) => {
+			const [ title, url ] = row.dimensions;
+			const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+				offsetDays: DATE_RANGE_OFFSET,
+			} ) );
+			const serviceURL = useSelect( ( select ) => select( STORE_NAME ).getServiceReportURL( 'content-pages', {
+				'explorer-table.plotKeys': '[]',
+				'_r.drilldown': `analytics.pagePath:${ url }`,
+				...generateDateRangeArgs( dateRange ),
+			} ) );
+			return (
+				<Link
+					href={ serviceURL }
+					external
+					inherit
+				>
+					{ title }
+				</Link>
+			);
+		},
+	},
+	{
+		title: __( 'Earnings', 'google-site-kit' ),
+		description: __( 'Earnings', 'google-site-kit' ),
+		field: 'metrics.0.values.0',
+		Component: ( { fieldValue } ) => numFmt(
+			fieldValue,
+			currencyFormat,
+		),
+	},
+];
```

Comment:
`tableColumns` needs to be defined within the component as we do elsewhere - this way the localization works as expected and also we don't have to use the `setCurrencyFormat` approach.